zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/kotlin/InboundBridgeRewriter.kt

```diff
@@ -76,20 +76,24 @@ import org.jetbrains.kotlin.name.Name
  * zipline.set(
  *   "helloService",
  *   object : InboundBridge<EchoService>(EchoSerializersModule) {
- *     val service: EchoService = TestingEchoService("hello")
- *     val serializer_0 = EchoSerializersModule.serializer<EchoRequest>()
- *     val serializer_1 = EchoSerializersModule.serializer<EchoResponse>()
- *     override fun call(inboundCall: InboundCall): ByteArray {
- *       return when {
- *         inboundCall.funName == "echo" -> {
- *           inboundCall.result(
- *             serializer_1,
- *             service.echo(
- *               inboundCall.parameter(serializer_0)
- *             )
- *           )
+ *     override fun create(context: Context): InboundCallHandler {
+ *       return object : InboundCallHandler {
+ *         val serializer_0 = context.serializersModule.serializer<EchoRequest>()
+ *         val serializer_1 = context.serializersModule.serializer<EchoResponse>()
+ *         override val context: Context = context
+ *         override fun call(inboundCall: InboundCall): ByteArray {
+ *           return when {
+ *             inboundCall.funName == "echo" -> {
+ *               inboundCall.result(
+ *                 serializer_1,
+ *                 s.echo(
+ *                   inboundCall.parameter(serializer_0)
+ *                 )
+ *               )
+ *             }
+ *             else -> inboundCall.unexpectedFunction()
+ *           }
```

Comment:
The diff in this example is a high-level view of the inbound side of the PR. I’ve split the `InboundBridge` interface into two parts, `InboundBridge` and `InboundCallHandler`. The big difference between the two parts is that `InboundBridge` is not attached to a particular endpoint, but `InboundCallHandler` is.